packages/@react-aria/checkbox/src/useCheckbox.ts

```diff
@@ -54,7 +54,7 @@ export function useCheckbox(props: AriaCheckboxProps, state: ToggleState, inputR
   useEffect(() => {
     // indeterminate is a property, but it can only be set via javascript
     // https://css-tricks.com/indeterminate-checkboxes/
-    if (inputRef.current) {
+    if (inputRef.current && isIndeterminate !== undefined) {
```

Comment:
Sorry, I found a bug, this visually works, but the checkbox representation isn't quite right
https://reactspectrum.blob.core.windows.net/reactspectrum/77db69d1e43672b16fddb698cd7ecc436e459bc9/storybook/index.html?path=/story/checkbox--default&providerSwitcher-express=false
if you inspect the dom in Chrome and select the input element, then go to the console and type `$0.indeterminate` it should give you the current value of that property on the input

go to the storybook controls and turn on isIndeterminate, check the console again, the `$0.indeterminate` should now be true

click the controls reset button, it looks like a u-turn arrow in the top right of the controls panel

check `$0.indeterminate` again, it should be false, but it's true instead

I think we need to not do this if check, and instead change the line below to be

```
inputRef.current.indeterminate = !!isIndeterminate;
```

Reply:
Thanks for the feedback! It indeed appears that the current implementation doesn't properly reset the indeterminate state of the checkbox.
I'll update the implementation.